go/tasks/plugins/k8s/sagemaker/sagemaker.go

```diff
@@ -0,0 +1,581 @@
+	} else {
+		return nil, errors.Errorf("The plugin is unable to get event info for unknown task type {%v}", m.TaskType)
+	}
+
+	logger.Infof(ctx, "Getting event information for task type: [%v], job region: [%v], job name: [%v], "+
+		"job type in url: [%v], sagemaker link name: [%v]", m.TaskType, jobRegion, jobName, jobTypeInURL, sagemakerLinkName)
+
+	cwLogURL := fmt.Sprintf("https://%s.console.aws.amazon.com/cloudwatch/home?region=%s#logStream:group=/aws/sagemaker/TrainingJobs;prefix=%s;streamFilter=typeLogStreamPrefix",
+		jobRegion, jobRegion, jobName)
+	smLogURL := fmt.Sprintf("https://%s.console.aws.amazon.com/sagemaker/home?region=%s#/%s/%s",
+		jobRegion, jobRegion, jobTypeInURL, jobName)
+
+	taskLogs := []*core.TaskLog{
+		{
+			Uri:           cwLogURL,
+			Name:          "CloudWatch Logs",
+			MessageFormat: core.TaskLog_JSON,
+		},
+		{
+			Uri:           smLogURL,
+			Name:          sagemakerLinkName,
+			MessageFormat: core.TaskLog_UNKNOWN,
+		},
+	}
+
+	customInfoMap := make(map[string]string)
+
+	customInfo, err := utils.MarshalObjToStruct(customInfoMap)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pluginsCore.TaskInfo{
+		Logs:       taskLogs,
+		CustomInfo: customInfo,
+	}, nil
+}
+
+func getOutputs(ctx context.Context, tr pluginsCore.TaskReader, outputPath string) (*core.LiteralMap, error) {
+	tk, err := tr.Read(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if tk.Interface.Outputs != nil && tk.Interface.Outputs.Variables == nil {
+		logger.Warnf(ctx, "No outputs declared in the output interface. Ignoring the generated outputs.")
+		return nil, nil
+	}
+
+	// We know that for XGBoost task there is only one output to be generated
+	if len(tk.Interface.Outputs.Variables) > 1 {
+		return nil, fmt.Errorf("expected to generate more than one outputs of type [%v]", tk.Interface.Outputs.Variables)
+	}
+	op := createOutputLiteralMap(tk, outputPath)
+	return op, nil
+}
+
+func createOutputPath(prefix string) string {
+	return fmt.Sprintf("%s/hyperparameter_tuning_outputs", prefix)
+}
+
+func createModelOutputPath(prefix, bestExperiment string) string {
+	return fmt.Sprintf("%s/%s/output/model.tar.gz", createOutputPath(prefix), bestExperiment)
+}
+
+func (m awsSagemakerPlugin) GetTaskPhaseForTrainingJob(
+	ctx context.Context, pluginContext k8s.PluginContext, trainingJob *trainingjobv1.TrainingJob) (pluginsCore.PhaseInfo, error) {
+
+	logger.Infof(ctx, "Getting task phase for sagemaker training job [%v]", trainingJob.Status.SageMakerTrainingJobName)
+	info, err := m.getEventInfoForJob(ctx, trainingJob)
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
+	}
+
+	occurredAt := time.Now()
+
+	switch trainingJob.Status.TrainingJobStatus {
+	case trainingjobController.ReconcilingTrainingJobStatus:
+		logger.Errorf(ctx, "Job stuck in reconciling status, assuming retryable failure [%s]", trainingJob.Status.Additional)
+		// TODO talk to AWS about why there cannot be an explicit condition that signals AWS API call errors
```

Comment:
did we talk to AWS about this?

Reply:
don't we get this problem often, especially if we are having some permissions problem in the begining and then it works? But, I guess it is ok to start off. Maybe one thing we could do is, put a simple time, check, if we see this status, but its not been atleast a couple minutes then we ignore?

Reply:
@kumare3 from my experience during the past couple of weeks, every time I got this problem it was due to an actual error in the CRD. I can't recall if I had seen any error that was due to transient permission errors.